src/components/cards/remote-cards/RemoteCard.tsx

```diff
@@ -0,0 +1,254 @@
+import { Linking, View } from 'react-native';
+import React, { useCallback } from 'react';
+import { get } from 'lodash';
+
+import { Box, Stack, Text, Inline } from '@/design-system';
+import { Icon } from '@/components/icons';
+import { ButtonPressAnimation } from '@/components/animations';
+import { TrimmedCard, useRemoteCardContext } from './RemoteCardProvider';
+import { IS_IOS } from '@/env';
+import { useNavigation } from '@/navigation';
+import { Language } from '@/languages';
+import { useAccountSettings, useDimensions } from '@/hooks';
+import { BackgroundColor, TextColor } from '@/design-system/color/palettes';
+import { CustomColor } from '@/design-system/color/useForegroundColor';
+import { maybeSignUri } from '@/handlers/imgix';
+import { Media } from '@/components/Media';
+import { SheetActionButton } from '@/components/sheet';
+import { colors } from '@/styles';
+
+const GUTTER_SIZE = 40;
+
+const getKeyForLanguage = (key: string, object: any, language: Language) => {
+  if (!object) {
+    return '';
+  }
+
+  const objectOrPrimitive = get(object, key);
+  if (typeof objectOrPrimitive === 'undefined') {
+    return '';
+  }
+
+  if (objectOrPrimitive === null) {
+    return '';
+  }
+
+  if (objectOrPrimitive[language]) {
+    return objectOrPrimitive[language];
+  }
+
+  return objectOrPrimitive[Language.EN_US] ?? '';
+};
+
+type RemoteCardProps = {
+  card: TrimmedCard;
+};
+
+type CardItem = {
+  icon: string;
+  text: string;
+  color?: TextColor | CustomColor;
+};
+
+export const RemoteCard: React.FC<RemoteCardProps> = ({
+  card = {} as TrimmedCard,
+}) => {
+  const { navigate } = useNavigation();
+  const { language } = useAccountSettings();
+  const { width } = useDimensions();
+  const { dismissCard } = useRemoteCardContext();
+
+  const { backgroundColor, primaryButton } = card;
+
+  const onDismissCard = useCallback(() => dismissCard(card.sys.id), [
+    card.sys.id,
+    dismissCard,
+  ]);
+
+  const onPress = useCallback(() => {
+    const internalNavigation = (route: string, options: any) => {
+      navigate(route, options);
+    };
+
+    if (primaryButton && primaryButton.url) {
+      Linking.openURL(primaryButton.url);
+    } else if (primaryButton && primaryButton.route) {
+      internalNavigation(primaryButton.route, primaryButton.props);
+    }
+  }, [navigate, primaryButton]);
+
+  const imageForPlatform = () => {
+    if (card.imageCollection.items.length === 1) {
+      return card.imageCollection.items[0].url;
```

Comment:
is it safe to access `card` w/o optional chaining? i see that `card` is cast as type `TrimmedCard`, but its default value is `{}`